seqr/views/utils/dataset_utils.py

```diff
@@ -444,19 +448,22 @@ def _load_rna_seq(model_cls, file_path, save_data, *args, user=None, **kwargs):
     individual_data_by_key = _get_individuals_by_key(projects)
 
     prev_loaded_individual_ids = set()
-    sample_guids_to_load = set()
+    sample_guid_keys_to_load = {}
     existing_samples_by_guid = {}
     samples_to_create = {}
 
     def update_sample_models():
         if samples_to_create:
-            _create_samples(
+            new_sample_models = _create_samples(
                 samples_to_create.values(),
                 user=user,
                 data_source=data_source,
                 sample_type=Sample.SAMPLE_TYPE_RNA,
                 dataset_type=Sample.DATASET_TYPE_VARIANT_CALLS,
             )
+            new_sample_ids = [s.id for s in new_sample_models]
+            sample_key_map = _get_sample_models_by_key(Sample.objects.filter(id__in=new_sample_ids), key_fields=key_fields)
+            sample_guid_keys_to_load.update({s['guid']: sample_key for sample_key, s in sample_key_map.items()})
```

Comment:
because guids are now generated after samples are actually bulk created in the database and not when the placeholder sample models are cretaed, we need to refactor how this code relies on sample guids